shared/profile/edit-profile/index.js

```diff
@@ -46,9 +47,9 @@ export default connect(
     return {
```

Comment:
How this works with the || is very problematic above. i ran into this when i was trying to make this use recompose and not use the store. We have a multiple sources of truth problem and really if want to keep this in the routeState we should have a way to load it once or something (ie if you go into it and delete your fullname it'll magically get replaced if this connect gets triggered, etc). So i think thats the root cause of this issue.

in terms of this code, you could also implement this entirely in the connect function and not change the onSumbit above. i'll make an example